From: Kir Kolyshkin <kolyshkin@gmail.com>
Date: Thu, 5 Jul 2018 11:51:50 -0700
Subject: [PATCH] loggerutils: build fixes, improve errors

There are two build errors when using go-1.11beta1:

> daemon/logger/loggerutils/logfile.go:367: Warningf format %q arg f.Name is a func value, not called
> daemon/logger/loggerutils/logfile.go:564: Debug call has possible formatting directive %v

In the first place, the file name is actually not required as error
message already includes it.

While at it, fix a couple of other places for more correct messages, and
make sure to not add a file name if an error already has it.

Fixes: f69f09f44c
Signed-off-by: Kir Kolyshkin <kolyshkin@gmail.com>
Signed-off-by: Sebastiaan van Stijn <github@gone.nl>
Origin: upstream, https://github.com/moby/moby/commit/09ad434
---
 daemon/logger/loggerutils/logfile.go | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- a/engine/daemon/logger/loggerutils/logfile.go
+++ b/engine/daemon/logger/loggerutils/logfile.go
@@ -333,7 +333,7 @@
 			if strings.HasSuffix(fileName, tmpLogfileSuffix) {
 				err := w.filesRefCounter.Dereference(fileName)
 				if err != nil {
-					logrus.Errorf("Failed to dereference the log file %q: %v", fileName, err)
+					logrus.Errorf("Failed to dereference log file %q: %v", fileName, err)
 				}
 			}
 		}
@@ -365,7 +365,7 @@
 			if strings.HasSuffix(f.Name(), tmpLogfileSuffix) {
 				err := os.Remove(f.Name())
 				if err != nil && !os.IsNotExist(err) {
-					logrus.Warningf("Failed to remove the logfile %q: %v", f.Name, err)
+					logrus.Warnf("Failed to remove logfile: %v", err)
 				}
 			}
 		}
@@ -437,7 +437,7 @@
 		rs.Close()
 		rErr := os.Remove(rs.Name())
 		if rErr != nil && !os.IsNotExist(rErr) {
-			logrus.Errorf("Failed to remove the logfile %q: %v", rs.Name(), rErr)
+			logrus.Errorf("Failed to remove logfile: %v", rErr)
 		}
 		return nil, errors.Wrap(err, "error while copying decompressed log stream to file")
 	}
@@ -562,7 +562,7 @@
 			}
 			return errRetry
 		case err := <-fileWatcher.Errors():
-			logrus.Debug("logger got error watching file: %v", err)
+			logrus.Debugf("logger got error watching file: %v", err)
 			// Something happened, let's try and stay alive and create a new watcher
 			if retries <= 5 {
 				fileWatcher.Close()
